docs/source/peer-chaincode-devmode.rst

```diff
@@ -0,0 +1,114 @@
+::
+    export FABRIC_CFG_PATH=$(pwd)/sampleconfig
+
+::
+
+Generate genesisblock.
+::
+    configtxgen -profile SampleDevModeSolo -channelID syschannel -outputBlock genesisblock -configPath $FABRIC_CFG_PATH -outputBlock $(pwd)/sampleconfig/genesisblock
+::
+
+Start the orderer
+-----------------
+
+::
+
+    ORDERER_GENERAL_GENESISPROFILE=SampleDevModeSolo orderer
+
+The above starts the orderer with the SampleDevModeSolo profile.
+
+Start the peer in dev mode
+--------------------------
+
+::
+
+    CORE_PEER_CHAINCODELISTENADDRESS=0.0.0.0:7052 peer node start --peer-chaincodedev=true
+
+The above command starts the peer using the default MSP.
+The ``--peer-chaincodedev=true`` puts it in “dev” mode.
+
+Create channel ch1
+---------------------------
+
+Generate the transactions for creating the channels using the ``configtxgen``
+tool with the SampleSingleMSPChannel profile.
+
+::
+
+   configtxgen -channelID ch1 -outputCreateChannelTx ch1.tx -profile SampleSingleMSPChannel -configPath $FABRIC_CFG_PATH
+
+::
+
+    peer channel create -o 127.0.0.1:7050 -c ch1 -f ch1.tx
+
+The above command assumes the orderer is reachable at ``127.0.0.1:7050``. The orderer
+is now tracking channel ch1 with the default configuration.
+
+::
+
+    peer channel join -b ch1.block
+
+The peer has now joined channel ch1.
+
+Build the chaincode
+-----------------------------
+
+The `simple` chaincode from the `integration/chaincode` directory is used here as an example.
+
+::
+    go build -o simpleChaincode ./integration/chaincode/simple/cmd
+::
+
+Start the chaincode
+-------------------
+
+::
+    CORE_CHAINCODE_LOGLEVEL=debug CORE_PEER_ADDRESS=192.168.0.7:7052 CORE_PEER_TLS_ENABLED=false CORE_CHAINCODE_ID_NAME=mycc:1.0 ./simpleChaincode -peer.address 192.168.0.7:7052
+
+The chaincode is started with the peer and chaincode logs indicating successful registration with the peer.
+- Note: The `CORE_CHAINCODE_ID_NAME` is set to `mycc:1.0` here, as when DevMode is enabled, lifecycle will automatically looking for launched chaincode with chaincodeID as `{ChaincodeName}:{ChaincodeVersion}`.
+
+Approve chaincode and commit chaincode
+--------------------------------------
+
+::
+    peer lifecycle chaincode approveformyorg  -o 127.0.0.1:7050 --channelID ch1 --name mycc --version 1.0 --sequence 1 --init-required --signature-policy "OR ('Org1MSP.peer','Org2MSP.peer')" --package-id {package-id}
+::
+
+::
+    peer lifecycle chaincode checkcommitreadiness -o 127.0.0.1:7050 --channelID ch1 --name mycc --version 1.0 --sequence 1 --init-required --signature-policy "OR ('Org1MSP.peer','Org2MSP.peer')"
+
+    peer lifecycle chaincode commit -o 127.0.0.1:7050 --channelID ch1 --name mycc --version 1.0 --sequence 1 --init-required --signature-policy "OR ('Org1MSP.peer','Org2MSP.peer')" --peerAddresses 192.168.0.7:7051
+::
+
+The above steps follow the new lifecycle commands which approve chaincode in both channels, and commit the chaincode to the ledger.
+
+
```

Comment:
Should there be a final line that mentions something like "at this point, you can invoke/query the chaincode as usual"? Seems like the doc just kinda ends here without actually talking about the user interacting with the chaincode.

Reply:
fixed.

Reply:
I was just going with what your previous text said, "peer AND chaincode logs." :)  Definitely correct it so it only mentions the log you had in mind.